### ISSUE TITLE
Dashboard: Grid navigation w/ Keyboard

### ISSUE THREAD

**Issue by BrittanyIRL** [GoogleForCreators/web-stories-wp#4296]
## Bug Description

Right now you have to tab through each action available on a grid item, which on a story is quite intensive since you have menus and buttons within each item. 

## Expected Behaviour

Tabbing through dashboard grids (stories or templates) should have the same functionality as the template detail gallery of template pages or the preview grid within the story editor. So, tab into the grid and then use arrow keys to navigate the grid items. If you hit tab again you exit the grid and move on through the page. 

## Steps to Reproduce

Tab through the dashboard grid, see that you focus on each focusable item inside of a grid item before moving to the next grid item and that using the keyboard to navigate grid items doesn't do anything. 

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

PR #4263 https://github.com/google/web-stories-wp/pull/4263 contains a reusable version of the editor's useGridViewKeys that isn't connected to context. This can be used to get pretty far on this fix, the added functionality is keying _into_ the grid item to interact with an item's focusable elements and them moving on - so adjusting tabIndex dynamically one extra layer. 

Fixing this issue should also fix #4205, #4198, #4201

**Comment by carloskelly13** [comment]
PR https://app.zenhub.com/workspace/o/google/web-stories-wp/issues/4317 has the commits for this ticket.
